packages/panzer/disc-fe/src/lof/Panzer_BlockedEpetraLinearObjFactory_impl.hpp

```diff
@@ -806,28 +817,65 @@ getThyraMatrix() const
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+Teuchos::RCP<const Thyra::VectorSpaceBase<double>>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
 getGhostedThyraDomainSpace() const
 {
-   if(ghostedDomainSpace_==Teuchos::null) {
-     if(colDOFManagerContainer_->containsBlockedDOFManager()) {
-       // loop over all vectors and build the vector space
-       std::vector<Teuchos::RCP<const Thyra::VectorSpaceBase<double> > > vsArray;
-       for(int i=0;i<getBlockColCount();i++)  
-          vsArray.push_back(Thyra::create_VectorSpace(getGhostedColMap(i)));
- 
-       ghostedDomainSpace_ = Thyra::productVectorSpace<double>(vsArray);
-     }
-     else {
-       // the domain space is not blocked (just an SPMD vector), build it from
-       // the zeroth column
-       ghostedDomainSpace_ = Thyra::create_VectorSpace(getGhostedColMap(0));
-     }
-   }
-   
-   return ghostedDomainSpace_;
-}
+  if (ghostedDomainSpace_ == Teuchos::null)
+  {
+    if (colDOFManagerContainer_->containsBlockedDOFManager())
+    {
+      // Loop over all vectors and build the vector space.
+      std::vector<Teuchos::RCP<const Thyra::VectorSpaceBase<double>>> vsArray;
+      for (int i(0); i < getBlockColCount(); ++i)
+        vsArray.push_back(Thyra::create_VectorSpace(getGhostedColMap(i)));
+      ghostedDomainSpace_ = Thyra::productVectorSpace<double>(vsArray);
+    }
+    else
+    {
+      // The domain space is not blocked (that is, we're just dealing with a
+      // SPMD vector), so build it from the zeroth column.
+      ghostedDomainSpace_ = Thyra::create_VectorSpace(getGhostedColMap(0));
+    }
+  }
+  return ghostedDomainSpace_;
+} // end of getGhostedThyraDomainSpace()
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  getGhostedThyraDomainSpace2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+Teuchos::RCP<const Thyra::VectorSpaceBase<double>>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
+getGhostedThyraDomainSpace2() const
+{
+  if (ghostedDomainSpace_ == Teuchos::null)
+  {
+    if (colDOFManagerContainer_->containsBlockedDOFManager())
+    {
+      // Loop over all vectors and build the vector space.
+      std::vector<Teuchos::RCP<const Thyra::VectorSpaceBase<double>>> vsArray;
+      for (int i(0); i < getBlockColCount(); ++i)
+        vsArray.push_back(Thyra::create_VectorSpace(getGhostedColMap2(i)));
+      ghostedDomainSpace_ = Thyra::productVectorSpace<double>(vsArray);
+    }
+    else
+    {
+      // The domain space is not blocked (that is, we're just dealing with a
+      // SPMD vector), so build it from the zeroth column.
+      ghostedDomainSpace_ = Thyra::create_VectorSpace(getGhostedColMap2(0));
+    }
+  }
+  return ghostedDomainSpace_;
+} // end of getGhostedThyraDomainSpace2()
```

Comment:
Version 2 of `getGhostedThyraDomainSpace` is the same as version 1, except it calls the version 2 `getGhostedColMap` routine.